src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -217,13 +217,19 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
 
         internal sealed override ScopedKind EffectiveScope => _scope;
 
-        internal sealed override bool UseUpdatedEscapeRules => _container?.UseUpdatedEscapeRules ?? false;
+        internal sealed override bool UseUpdatedEscapeRules =>
+            _container switch
+            {
+                MethodSymbol method => method.UseUpdatedEscapeRules,
+                Symbol symbol => symbol.ContainingModule.UseUpdatedEscapeRules,
```

Comment:
This is always a module-wide setting, right? So we're not worried about accidentally looking in the wrong place for the answer here. For example, if this parameter's `_container` were an indexer.

Reply:
Yes, this should always be from a module-level setting.